builder/make-config-files.nix

```diff
@@ -1,5 +1,7 @@
 { stdenv, lib, haskellLib, ghc, nonReinstallablePkgs, runCommand, writeText, writeScript }:
 
+{ identifier, component, fullName, flags ? {}, needsProfiling ? false, chooseDrv ? drv: drv }:
```

Comment:
This feels like a bit of a hack - what goes wrong if we say we "depend on" the main component? Isn't the package-db stuff the same between the main one and the haddock one?

Reply:
> Isn't the package-db stuff the same between the main one and the haddock one?

The paths `haddock-interfaces:` fields in the `.conf` files are replaced so that the generated hyperlinks will work.  They have to point to the real locations of the file generated haddock files for haddock to work.